src/features/groups/components/ContainerCardList/ContainerCard/RenameContainerForm.tsx

```diff
@@ -0,0 +1,118 @@
+import { ContainerIcon } from '@/assets/images/icons';
+import { Card, Icon, SquareTextInput } from '@/components/ui';
+import { Form, FormControl, FormField, FormItem, FormMessage } from '@/components/ui/Form';
+import { renameContainerFormSchema, RenameContainerInputs } from '@/features/groups/lib/schemas';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { KeyboardEvent, useEffect, useRef } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+interface IRenameContainerFormProps {
+  /**
+   * The current container name, which a user willing to change from
+   */
+  currentContainerName: string;
+  /**
+   * The state, if true, the rename form is open
+   */
+  isOpen: boolean;
+  /**
+   * The function to change the state of form to false (=close the input field)
+   */
+  onClose: () => void;
+}
+
+export const RenameContainerForm = ({
+  currentContainerName,
+  isOpen,
+  onClose,
+}: IRenameContainerFormProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const form = useForm<z.infer<typeof renameContainerFormSchema>>({
+    resolver: zodResolver(renameContainerFormSchema),
+    defaultValues: {
+      containerName: currentContainerName,
+    },
+  });
+
+  const processSubmit: SubmitHandler<RenameContainerInputs> = async (
+    values: RenameContainerInputs,
+  ) => {
+    const { containerName } = values;
+    if (containerName === currentContainerName) return;
+  };
+
+  /**
+   * Handle the key down event.
+   * when the escape key is pressed, close the form without the submission.
+   * @param event - The keyboard event
+   */
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      onClose();
+    }
+  };
+
+  /**
+   * Clear the input value.
+   */
+  const handleClearInput = () => {
+    form.setValue('containerName', '');
+  };
+
+  /**
+   * Handle the outside click event.
+   * process the form submission and close the form.
+   */
+  const handleOutsideClick = () => {
+    // process the form submission
+    processSubmit(form.getValues());
+    onClose();
+  };
+
+  useEffect(() => {
+    if (!isOpen) form.reset();
+  }, [isOpen, form]);
+
+  useEffect(() => {
+    if (isOpen) {
+      inputRef.current?.focus();
+    }
+  }, [isOpen]);
+
+  return (
+    <Card asChild>
+      <div className="flex gap-2 items-center px-4 py-2">
+        <div className="flex items-center justify-center bg-accent rounded-full min-w-11 h-11">
```

Comment:
May I hear why chose `min-w-11`? If there is no special reason, `w-11` should be appropriate.